### PR TITLE
Check file system validity before reading opening.bnr

### PR DESCRIPTION
**Pull request by JosJuice**

### CODE REVIEW

**Comment by MaJoRoesch** [comment]
No description? Well, I have my suspicions, but can you say what this means for users?

**Comment by JosJuice** [comment]
If the partition is invalid (because the GC/Wii magic word can't be found) and GetNames is called, Dolphin will crash without this change. Now that I think about it, that case can't be triggered right now since the magic word not being present means that the volume can't even be created. I ran into this when testing PR #2353, which changes the file system code to check for the magic word within the relevant partition, while the volume code still checks for the magic word in the unencrypted header.

Should I close this and make it part of PR #2353? As a separate PR, it makes the code a bit cleaner, but it shouldn't actually affect anything for users.

**Comment by MaJoRoesch** [comment]
Thanks for describing it! But apparently I was editing the comment while you were responding to it. >_< What do you think of that idea?

**Comment by JosJuice** [comment]
It would be possible to have such a check, but it doesn't fit in GetNames. If a warning is displayed, it should be displayed when the game is launched or when ISOProperties is opened, not when the game list is scanned. Besides, if we want to check for bad dumps, there's a lot more things we should check in headers than the presence of these four bytes.

**Comment by JosJuice** [comment]
For the record, I don't even know if the magic word needs to be set correctly for games to boot. Games should already know if they have GC or Wii file systems, unlike Dolphin.

File: Source/Core/DiscIO/VolumeWiiCrypted.cpp [hunk 1]
```diff
@@ -205,6 +205,9 @@ std::string CVolumeWiiCrypted::GetInternalName() const
 std::map<IVolume::ELanguage, std::string> CVolumeWiiCrypted::GetNames() const
 {
 	std::unique_ptr<IFileSystem> file_system(CreateFileSystem(this));
+	if (file_system == nullptr)
+		return std::map<IVolume::ELanguage, std::string>();
```
Comment (lioncash) [inline Source/Core/DiscIO/VolumeWiiCrypted.cpp]: You can just do `return {};` to return an empty map. Also you can just use `!` to check for nullity of the unique_ptrs.

**Comment by JosJuice** [comment]
I've decided to include this fix in the PRs that can make the problem occur (#2353 and #2820) instead of having a separate PR.

**Pull request closed without merging**